python/openinference-instrumentation/src/openinference/instrumentation/__init__.py

```diff
@@ -166,6 +184,61 @@ def __init__(self, tags: List[str]) -> None:
         super().__init__(tags=tags)
 
 
+class using_prompt_template(_UsingAttributesContextManager):
+    """
+    Context manager to add prompt template to the current OpenTelemetry Context. OpenInference
+    instrumentations will read this Context and pass the prompt template as a span attribute,
+    following the OpenInference semantic conventions.
+
+    Examples:
+        prompt_template = "Please describe the weather forecast for {city} on {date}"
+        with using_prompt_template(prompt_template):
+            # Tracing within this block will include the span attribute:
+            # "llm.prompt_template.template" = "Please describe the weather
+            forecast for {city} on {date}"
+            ...
+    """
+
+    def __init__(self, prompt_template: str) -> None:
+        super().__init__(prompt_template=prompt_template)
+
+
+class using_prompt_template_version(_UsingAttributesContextManager):
+    """
+    Context manager to add prompt template version to the current OpenTelemetry Context.
+    OpenInference instrumentations will read this Context and pass the prompt template
+    version as a span attribute, following the OpenInference semantic conventions.
+
+    Examples:
+        with using_prompt_template_version("v1.0"):
+            # Tracing within this block will include the span attribute:
+            # "llm.prompt_template.version" = "v1.0"
```

Comment:
Let's combine all of these as they should probably all be set at the same time

Reply:
Much better, what was I thinking?